evalml/tests/component_tests/test_featuretools.py

```diff
@@ -163,3 +164,55 @@ def test_dfs_sets_max_depth_1(mock_dfs, X_y_multi):
     feature.fit(X_pd, y)
     _, kwargs = mock_dfs.call_args
     assert kwargs["max_depth"] == 1
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+def test_dfs_with_serialized_features(mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+    X_fit = X_pd.iloc[: len(X) // 3]
+    X_transform = X_pd.iloc[len(X) // 3 :]
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_transform, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(
+        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
+    )
+
+    feature = DFSTransformer(features=features)
+    feature.fit(X_fit)  # no-op
+    assert not mock_dfs.called
+
+    X_t = feature.transform(X_transform)
+    assert_frame_equal(feature_matrix, X_t)
+    assert features == feature.features
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+@patch(
+    "evalml.pipelines.components.transformers.preprocessing.featuretools.calculate_feature_matrix"
+)
+def test_dfs_skip_transform(mock_calculate_feature_matrix, mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+    X_fit = X_pd.iloc[: len(X) // 3]
+    X_transform = X_pd.iloc[len(X) // 3 :]
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_transform, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(entityset=es, target_dataframe_name="X")
+
+    feature = DFSTransformer(features=features)
+    feature.fit(X_fit)  # no-op
+    X_t = feature.transform(feature_matrix)  # no-op as well
+    assert not mock_dfs.called
+    assert not mock_calculate_feature_matrix.called
+
+    assert_frame_equal(feature_matrix, X_t)
+    assert features == feature.features
```

Comment:
Can we add a test for:
- Passing in features such that feature.get_feature_names() doesn't exist in the dataframe